content/chainguard/chainguard-enforce/authentication/custom-idps/index.md

```diff
@@ -0,0 +1,194 @@
+
+```
+default:
+  identity-provider: <your identity provider id here>
+```
+
+Then save and close the file. If you used the default editor, `nano`, you can do so by pressing `CTRL + X`, `Y`, and then `ENTER`. 
+
+Once set, the configured identity provider will be used automatically any time you run `chainctl auth login`.
+
+> **Note**: `chainctl auth login --headless` is not currently compatible with SSO authentication.
+
+
+### Authenticate with the Chainguard Enforce Console
+
+To authenticate with the Chainguard Enforce Consle using SSO, click the **Use your identity provider** link on the login page.
+
+![Screenshot showing an example Chainguard login page, with a red arrow pointing to the "Use your identity provider" link.](sso-1.png)
+
+On the next page, add the identity provider's unique ID.
+
+![Screenshot showing an example Chainguard login page with a field reading "IDP Identifier"](sso-2.png)
+
+After adding your ID, click the **Continue** button. You'll then be redirected to your identity provider to authenticate, after which you'll be redirected back to the Console.
+
+
+## Setup and Administration
+
+Chainguard SSO requires OpenID Connect (OIDC) compatible identity providers. In particular, identity providers must also support the following.
+
+* The `authorization code` grant type (sometimes called the `authorization code` *flow*).
+* The standard `openid`, `email` and `profile` scopes. Note that the Chainguard platform will partially function with only the `openid` scope, but full functionality requires the `email` and `profile` scopes as well.
+
+Customer-managed identity providers must also have a public, unauthenticated OIDC discovery endpoint.
+
+In order to set up SSO for your identity provider, you must configure an OIDC application that the Chainguard platform can use on your identity provider. Following that, you have to configure the Chainguard platform to use that application.
+
+
+### Integration Guides for supported identity providers
+
+If your identity provider is Okta, Ping Identity or Azure Active Directory, we’ve published step-by-step integration guides for your platform.
+
+* [Okta](/chainguard/chainguard-enforce/authentication/example-idps/okta/)
+* [Ping Identity](/chainguard/chainguard-enforce/authentication/example-idps/ping-id/)
+* [Azure Active Directory](/chainguard/chainguard-enforce/authentication/example-idps/azure-ad/)
+
+If you aren't using one of these identity providers, you can complete the following Generic Integration Guide to configure your provider to work with Chainguard Enforce. However, be aware that Chainguard does not actively support identity providers other than the ones listed previously. If you'd like to request that we support a specific identity provider, we encourage you to [contact us](https://www.chainguard.dev/contact?utm_source=docs) to do so.
```

Comment:
I think this is good, I would maybe even make fewer claims, like below, but feel free to finesse further!

```suggestion
If you aren't using one of these identity providers, you can complete the following Generic Integration Guide to configure your provider to work with Chainguard Enforce. However, be aware that Chainguard does not actively support identity providers other than the ones listed previously. If you are using an alternate identity provider, we encourage you to [contact us](https://www.chainguard.dev/contact?utm_source=docs) to learn more.
```